### PR TITLE
fix issue with null byte string by dumping it first

### LINKED ISSUE #85: string contains null byte
#14 has reappeared when using nokogiri 1.17.2.

Try it with RxSwift and I got the following output

```
string contains null byte
/.gem/ruby/3.3.0/gems/nokogiri-1.17.2-arm64-darwin/lib/nokogiri/xml/document.rb:307:in `new'
/.gem/ruby/3.3.0/gems/nokogiri-1.17.2-arm64-darwin/lib/nokogiri/xml/document.rb:307:in `create_cdata'
/.gem/ruby/3.3.0/gems/nokogiri-1.17.2-arm64-darwin/lib/nokogiri/xml/builder.rb:346:in `cdata'
/.gem/ruby/3.3.0/gems/cyclonedx-cocoapods-1.4.1/lib/cyclonedx/cocoapods/bom_builder.rb:130:in `block (2 levels) in add_to_bom'
```

### PR DESCRIPTION
**Pull request by fnxpt**
Fixes #85

### CODE REVIEW

File: lib/cyclonedx/cocoapods/bom_builder.rb [hunk 1]
```diff
@@ -127,7 +127,7 @@ def add_to_bom(xml, manifest_path, trim_strings_length = 0)
           xml_add_author(xml, trim_strings_length)
           xml.name_ name
           xml.version version.to_s
-          xml.description { xml.cdata description } unless description.nil?
+          xml.description { xml.cdata description.dump[1..-2] } unless description.nil?
```
Comment (macblazer) [inline lib/cyclonedx/cocoapods/bom_builder.rb:130]: Might be nice to add a comment on the line above this something along the lines of

```
# Use `dump` to escape non-printing characters, then remove the starting and trailing double-quotes from `dump`.
```

Also, would be nice to have a unit test that sets up a pod with a description that contains a null byte in the middle so we can be sure this continues to solve the problem.

**Review by macblazer — CHANGES_REQUESTED** [review]

**Comment by jkowalleck** [comment]
> Codacy complaint, but the code looks good semantically. Thanks for adding the unit test.

codacy says: 
> https://app.codacy.com/gh/CycloneDX/cyclonedx-cocoapods/pull-requests/86/issues#issue-85bd871545c6bc02d711a60bd6884ec5
> Line is too long. [122/120]

**Comment by fnxpt** [comment]
Can we merge this?
